executions/graphql-kotlin-transaction-batcher-instrumentation/src/main/kotlin/com/expediagroup/graphql/transactionbatcher/instrumentation/state/ExecutionState.kt

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2022 Expedia, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.expediagroup.graphql.transactionbatcher.instrumentation.state
+
+import graphql.schema.DataFetcher
+
+enum class LevelState { NOT_DISPATCHED, DISPATCHED }
+
+/**
+ * Handle the state of an [graphql.ExecutionInput]
+ */
+class ExecutionState(documentHeight: Int) {
```

Comment:
maybe rename to `BatchExecutionState`?

Reply:
i would keep it as `ExecutionState` an instance of this class is associated with to an instance of `ExecutionInput`, in theory this Instrumentation can also apply to a single ExecutionInput (single GraphQLRequest), as it kind of covers what the DataLoaderInstrumentation does, it will batch operations even for a single Execution.

Reply:
it is still related to batching even within single request so I think `ExecutionBatchState` is valid and more meaningful